src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Button, Notification } from 'hds-react';
+import { useHistory } from 'react-router-dom';
+import formatDate from 'date-fns/format';
+import parse from 'date-fns/parse';
+import api, { Resource } from '../../common/utils/api/api';
+import { ResourceInfo } from '../../resource/page/ResourcePage';
+import Datepicker from '../../components/datepicker/Datepicker';
+import './CreateNewOpeningPeriodPage.scss';
+import { ResourceState } from '../../common/lib/types';
+
+type Inputs = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string;
+  openingPeriodEndDate: string;
+};
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+export default function CreateNewOpeningPeriodPage({
+  resourceId,
+}: {
+  resourceId: string;
+}): JSX.Element {
+  const { register, handleSubmit, errors } = useForm<Inputs>({
+    mode: 'all',
+    defaultValues: {
+      openingPeriodTitle: '',
+      openingPeriodOptionalDescription: '',
+      openingPeriodBeginDate: undefined,
+      openingPeriodEndDate: undefined,
+    },
+  });
+  const history = useHistory();
+  const [resource, setResource] = useState<Resource>();
+  const [hasLoadingResourceError, setLoadingResourceError] = useState<
+    Error | undefined
+  >(undefined);
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [periodBeginDate, setPeriodBeginDate] = useState<Date | null>(null);
+  const [periodEndDate, setPeriodEndDate] = useState<Date | null>(null);
+  const [submitStatus, setSubmitStatus] = useState<SubmitStatus>('init');
+
+  const onSubmit = async (data: Inputs): Promise<void> => {
+    try {
+      if (!resource) {
+        throw new Error('This should never happen, but typescript');
+      }
+      await api.postDatePeriod({
+        resource: resource.id,
+        name: {
+          fi: data.openingPeriodTitle,
+          sv: null,
+          en: null,
+        },
+        description: {
+          fi: data.openingPeriodOptionalDescription,
+          sv: null,
+          en: null,
+        },
+        start_date: formatDate(
+          parse(data.openingPeriodBeginDate, 'dd.MM.yyyy', new Date()),
+          'yyyy-MM-dd'
+        ),
+        end_date: formatDate(
+          parse(data.openingPeriodEndDate, 'dd.MM.yyyy', new Date()),
+          'yyyy-MM-dd'
+        ),
```

Comment:
The formats could locate in from common/utils/date-time

Reply:
- [ ] Move these formats under common/utils/date-time